cip/1.accepted/CIP2017-03-29-Single-Value-Subqueries.adoc

```diff
@@ -0,0 +1,68 @@
+= CIP2017-03-29 Scalar Subqueries and List Subqueries
+:numbered:
+:toc:
+:toc-placement: macro
+:source-highlighter: codemirror
+
+*Author:* Tobias Lindaaker <tobias.lindaaker@neotechnology.com>
+
+toc::[]
+
+== Scalar Subqueries and List Subqueries
+
+Scalar Subqueries are read-only subqueries that produce a single value in a single row.
+The result of a Scalar Subquery is the single value (in the single row) produced by the subquery.
+
+List Subqueries are read-only subqueries that produce a single value per row, and zero or more rows.
+The result of a List Subquery is to collect the value of all rows produced by the subquery into a list.
+
+=== Syntax
+
+[source, ebnf]
+----
+ListSubquery = '[', SingleValueSubquery, ']' ;
+
+ScalarSubquery = 'SCALAR', '(', SingleValueSubquery, ')' ;
+
+SingleValueSubquery = SingleValuePatternSubquery
+                    | SingleValueUnwindSubquery
+                    | SingleValueReadSubquery
+                    ;
+
+SingleValuePatternSubquery = PatternPart, Filter, SingleValueReturn ;
+SingleValueUnwindSubquery = 'UNWIND', Expression, ['AS', Variable, Filter] ;
```

Comment:
The implication here is that the implicitly `RETURN` value is the value unpacked by `UNWIND`.

`UNWIND` with `RETURN` is handled by the regular `SingleValueReadSubquery`.

The use cases for this form is:

1. Unpacking a singleton-list: `SCALAR( UNWIND myList )`
2. Sorting a list: `[UNWIND myList AS x ORDER BY x.age]`
3. Filtering a list: `[UNWIND myList AS x WHERE x.name CONTAINS ' ']`

et.c.